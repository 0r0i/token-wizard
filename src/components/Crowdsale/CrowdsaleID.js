import React from 'react'
import { ButtonCopyToClipboard } from '../Common/ButtonCopyToClipboard'

export const CrowdsaleID = ({ hash, description, extraClassname = '' }) => {
  return (
    <div className={`cs-CrowdsaleID ${extraClassname}`}>
      <div className="cs-CrowdsaleID_InnerContainer">
        <h3 className="cs-CrowdsaleID_Hash">
          <span className="cs-CrowdsaleID_HashText">{hash}</span> <ButtonCopyToClipboard value={hash} />
        </h3>
        <p className="cs-CrowdsaleID_Description">{description}</p>
      </div>
    </div>
  )
}
