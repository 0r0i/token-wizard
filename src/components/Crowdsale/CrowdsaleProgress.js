import React from 'react'

export const CrowdsaleProgress = ({ totalRaisedFunds, ethGoal, tokensClaimedRatio, extraClassName = '' }) => {
  const renderDividers = () => {
    return Array.apply('', Array(9)).map((noUse, index) => (
      <div key={index} className="cs-CrowdsaleProgress_Division" />
    ))
  }

  return (
    <div className={`cs-CrowdsaleProgress ${extraClassName}`}>
      <div className="cs-CrowdsaleProgress_Funds">
        <div className="cs-CrowdsaleProgress_FundsAmount cs-CrowdsaleProgress_FundsAmount-total-raised">
          <h2 className="cs-CrowdsaleProgress_FundsTitle">
            {totalRaisedFunds} <span className="cs-CrowdsaleProgress_NoBreak">ETH</span>
          </h2>
          <p className="cs-CrowdsaleProgress_FundsDescription">Total Raised Funds</p>
        </div>
        <div className="cs-CrowdsaleProgress_FundsAmount cs-CrowdsaleProgress_FundsAmount-goal">
          <h2 className="cs-CrowdsaleProgress_FundsTitle">
            {ethGoal} <span className="cs-CrowdsaleProgress_NoBreak">ETH</span>
          </h2>
          <p className="cs-CrowdsaleProgress_FundsDescription">Goal</p>
        </div>
      </div>
      <div className="cs-CrowdsaleProgress_Bar">
        <div className="cs-CrowdsaleProgress_BarWidth" style={{ width: tokensClaimedRatio + '%' }} />
        {renderDividers()}
      </div>
    </div>
  )
}
