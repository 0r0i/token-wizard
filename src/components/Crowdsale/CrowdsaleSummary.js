import React from 'react'

export const CrowdsaleSummary = ({
  contributorsCount,
  currentRatePerETH,
  endRatePerETH,
  extraClassName = '',
  isMintedCappedCrowdsale,
  startRatePerETH,
  tokensClaimed,
  totalSupply
}) => {
  const isDutchAuctionCrowdsale = !isMintedCappedCrowdsale

  const renderCrowdsaleSummaryItem = (title, description) => {
    return (
      <div className="cs-CrowdsaleSummary_Item">
        <h3 className="cs-CrowdsaleSummary_ItemTitle">{title}</h3>
        <p className="cs-CrowdsaleSummary_ItemDescription">{description}</p>
      </div>
    )
  }

  return (
    <div className={`cs-CrowdsaleSummary ${extraClassName}`}>
      {renderCrowdsaleSummaryItem(tokensClaimed, 'Tokens Claimed')}
      {renderCrowdsaleSummaryItem(contributorsCount, 'Contributors')}
      {isMintedCappedCrowdsale ? renderCrowdsaleSummaryItem(currentRatePerETH, 'Rate (Tokens/ETH)') : null}
      {renderCrowdsaleSummaryItem(totalSupply, 'Total Supply')}
      {isDutchAuctionCrowdsale ? renderCrowdsaleSummaryItem(startRatePerETH, 'Start Rate (Tokens/ETH)') : null}
      {isDutchAuctionCrowdsale ? renderCrowdsaleSummaryItem(currentRatePerETH, 'Rate (Tokens/ETH)') : null}
      {isDutchAuctionCrowdsale ? renderCrowdsaleSummaryItem(endRatePerETH, 'End Rate (Tokens/ETH)') : null}
    </div>
  )
}
