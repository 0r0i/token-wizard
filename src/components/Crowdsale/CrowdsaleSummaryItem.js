import React from 'react'

export const CrowdsaleSummaryItem = ({ title, description }) => {
  return (
    <div className="cs-CrowdsaleSummaryItem">
      <h3 className="cs-CrowdsaleSummaryItem_Title">{title}</h3>
      <p className="cs-CrowdsaleSummaryItem_Description">{description}</p>
    </div>
  )
}
