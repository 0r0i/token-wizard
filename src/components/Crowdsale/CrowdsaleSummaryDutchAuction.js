import React from 'react'
import { CrowdsaleSummaryItem } from './CrowdsaleSummaryItem'

export const CrowdsaleSummaryDutchAuction = ({
  contributorsCount,
  currentRatePerETH,
  endRatePerETH,
  extraClassName = '',
  startRatePerETH,
  tokensClaimed,
  totalSupply
}) => {
  return (
    <div className={`cs-CrowdsaleSummaryDutchAuction ${extraClassName}`}>
      <CrowdsaleSummaryItem title={tokensClaimed} description={'Tokens Claimed'} />
      <CrowdsaleSummaryItem title={contributorsCount} description={'Contributors'} />
      <CrowdsaleSummaryItem title={totalSupply} description={'Total Supply'} />
      <CrowdsaleSummaryItem title={startRatePerETH} description={'Start Rate (Tokens/ETH)'} />
      <CrowdsaleSummaryItem title={currentRatePerETH} description={'Rate (Tokens/ETH)'} />
      <CrowdsaleSummaryItem title={endRatePerETH} description={'End Rate (Tokens/ETH)'} />
    </div>
  )
}
