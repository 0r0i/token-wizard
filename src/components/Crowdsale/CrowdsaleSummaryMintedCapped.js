import React from 'react'
import { CrowdsaleSummaryItem } from './CrowdsaleSummaryItem'

export const CrowdsaleSummaryMintedCapped = ({
  contributorsCount,
  currentRatePerETH,
  extraClassName = '',
  tokensClaimed,
  totalSupply
}) => {
  return (
    <div className={`cs-CrowdsaleSummaryMintedCapped ${extraClassName}`}>
      <CrowdsaleSummaryItem title={tokensClaimed} description={'Tokens Claimed'} />
      <CrowdsaleSummaryItem title={contributorsCount} description={'Contributors'} />
      <CrowdsaleSummaryItem title={currentRatePerETH} description={'Rate (Tokens/ETH)'} />
      <CrowdsaleSummaryItem title={totalSupply} description={'Total Supply'} />
    </div>
  )
}
