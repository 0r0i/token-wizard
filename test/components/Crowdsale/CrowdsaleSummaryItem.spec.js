import React from 'react'
import render from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, shallow } from 'enzyme'
import { CrowdsaleSummaryItem } from '../../../src/components/Crowdsale/CrowdsaleSummaryItem'

configure({ adapter: new Adapter() })

describe(`CrowdsaleSummaryItem`, () => {
  it(`should contain an element to display the title's text`, () => {
    const theTitle = 'The title'
    const wrapper = shallow(<CrowdsaleSummaryItem title={theTitle} />)
    const itemTitle = wrapper.find('.cs-CrowdsaleSummaryItem_Title')

    expect(itemTitle.exists()).toBeTruthy()
    expect(itemTitle.text()).toBe(theTitle)
  })

  it(`should contain an element to display the description's text`, () => {
    const theDescription = "The item's description"
    const wrapper = shallow(<CrowdsaleSummaryItem description={theDescription} />)
    const itemDescription = wrapper.find('.cs-CrowdsaleSummaryItem_Description')

    expect(itemDescription.exists()).toBeTruthy()
    expect(itemDescription.text()).toBe(theDescription)
  })

  it(`should render CrowdsaleSummaryItem component`, () => {
    const theTitle = 'The title'
    const theDescription = "The item's description"

    // Given
    const component = render.create(<CrowdsaleSummaryItem title={theTitle} description={theDescription} />)

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })
})
