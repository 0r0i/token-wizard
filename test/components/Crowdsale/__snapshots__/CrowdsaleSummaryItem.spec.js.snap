// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CrowdsaleSummaryItem should render CrowdsaleSummaryItem component 1`] = `
<div
  className="cs-CrowdsaleSummaryItem"
>
  <h3
    className="cs-CrowdsaleSummaryItem_Title"
  >
    The title
  </h3>
  <p
    className="cs-CrowdsaleSummaryItem_Description"
  >
    The item's description
  </p>
</div>
`;
